### PR TITLE
[R] Autogeneration of R model wrappers

### PR DESCRIPTION
**Pull request by terrytangyuan**
* Added utility function to generate model wrappers from Python API
* Added script to generate all the wrappers to `auto_generated_wrappers.R` and regenerate NAMESPACE/Rd files
* Update reticulate version requirement
* Added all remaining model wrappers: gradient boosting, GLM, PCA, and SVD
* Made sure all existing unit tests pass

### CODE REVIEW

**Comment by navdeep-G** [comment]
@terrytangyuan How do you feel about adding GLM, PCA, and SVD to this PR?

File: src/interface_r/DESCRIPTION [hunk 1]
```diff
@@ -19,7 +19,7 @@ Depends:
     R (>= 3.1)
 Imports:
     magrittr,
-    reticulate (>= 1.3.1.9000)
+    reticulate
```
[the remark this answers is not in the document]
Comment (terrytangyuan) [reply src/interface_r/DESCRIPTION]: Actually I will change this back. This currently relies on Github dev version of reticulate.

**Comment by terrytangyuan** [comment]
@navdeep-G Yea I can certainly add those if all other pieces in this PR look good.

**Comment by navdeep-G** [comment]
@terrytangyuan Yes, the rest looks good to me. Thoughts @ledell 

It would be good to have the relevant estimators in. I can see you only need GLM, PCA, and SVD left. So once those are in we can start testing out the package more thoroughly.

**Comment by terrytangyuan** [comment]
@navdeep-G Added the estimators you mentioned. Wanna approve this and then I'll merge?

**Comment by ledell** [comment]
@navdeep-G @terrytangyuan I am noticing that we slightly diverge from the H2O4GPU Python naming scheme here in the GLM by using "Regressor" instead of "Regression" (which is what sklearn uses, even though it's internally inconsistent).  I think this is a good change.

**Comment by ledell** [comment]
@terrytangyuan One thing I a just noticing now is that you're using a single string value as a default instead of a vector of strings when there's a fixed list of options.  Example: `solver = "liblinear"` in `h2o4gpu.logistic_regressor()`. 

Since we only have a single default string value in Python H2O4GPU, there is obviously no easy way to autogenerate this list of values for R.  So I am wondering if there some way that we can encode the allowable string options in Python (other than listing the options in the docstring)?

**Comment by navdeep-G** [comment]
@ledell @terrytangyuan One issue here is the family should NOT be changed for logistic_regressor() since its a logistic regression solvers. It seems reticulate is picking up from the constructor of implementation, which has `solver="liblinear"`. We might need to hide these options from the function itself since it shouldn't be changed.

**Comment by terrytangyuan** [comment]
Good catch! If we somehow need to automate that, we can certainly do it in some hacky way. We can store constant classes and everytime when we see a string, we can search for its existence in the constant classes and use the list of possible strings in the constant class that single string value belongs to.

**Comment by ledell** [comment]
@navdeep-G Ah ok, very interesting.  That brings up a bigger issue.  So maybe `solver` should not be part of the  `LogisticRegression` constructor in Python at all?  https://github.com/h2oai/h2o4gpu/blob/master/src/interface_py/h2o4gpu/solvers/logistic.py#L37

**Comment by navdeep-G** [comment]
@ledell Yes, it should not be. I will make an issue and assign to myself.

**Comment by navdeep-G** [comment]
Made it here: https://github.com/h2oai/h2o4gpu/issues/395

**Pull request merged**